pytorch_lightning/accelerators/accelerator.py

```diff
@@ -232,7 +232,7 @@ def sync_tensor(self,
         Return:
             reduced value
         """
-        raise NotImplementedError()
```

Comment:
this error has the advantage that new accelerators are forced to implement it, because a default behavior (like no op) is not well defined. 
a possible alternative is to apply this fix directly in cpuaccelerator, and implement the no op there explicitly.

Reply:
I agree with this, after thinking about this more it's important to enforce this, and add the explicit override throughout accelerators rather than assume no-op. Made the change